[Bug] Deck Export fails if name contains apostrophe

Message: i tried to export a deck but an error happen that won't let me do it

```
android.database.sqlite.SQLiteException: error while compiling: "ATTACH '/storage/emulated/0/Android/data/com.ichi2.anki/cache/export/Life's skill-20230421181612.anki2' AS DST_DB": DbError { info: "SqlInputError { error: Error { code: Unknown, extended_code: 1 }, msg: \"near \\\"s\\\": syntax error\", sql: \"ATTACH '/storage/emulated/0/Android/data/com.ichi2.anki/cache/export/Life's skill-20230421181612.anki2' AS DST_DB\", offset: 74 }", kind: Other }
	at net.ankiweb.rsdroid.database.StreamingProtobufSQLiteCursor.<init>(StreamingProtobufSQLiteCursor.kt:51)
	at net.ankiweb.rsdroid.database.RustSupportSQLiteDatabase.query(RustSupportSQLiteDatabase.kt:2)
	at net.ankiweb.rsdroid.database.RustSupportSQLiteDatabase.execSQL(RustSupportSQLiteDatabase.kt:2)
	at net.ankiweb.rsdroid.database.RustSupportSQLiteDatabase.execSQL(RustSupportSQLiteDatabase.kt:1)
	at com.ichi2.utils.DatabaseChangeDecorator.execSQL(DatabaseChangeDecorator.kt:1)
	at com.ichi2.libanki.AnkiExporter.exportInto(AnkiPackageExporter.kt:89)
	at com.ichi2.libanki.AnkiPackageExporter.exportFiltered(AnkiPackageExporter.kt:13)
	at com.ichi2.libanki.AnkiPackageExporter.exportInto(AnkiPackageExporter.kt:67)
	at com.ichi2.anki.export.ActivityExportingDelegate$exportApkgOrColpkgLegacy$1$1$1.invoke(ActivityExportingDelegate.kt:5)
	at com.ichi2.anki.export.ActivityExportingDelegate$exportApkgOrColpkgLegacy$1$1$1.invoke(ActivityExportingDelegate.kt:1)
	at com.ichi2.anki.CollectionManager$withCol$2.invoke(CollectionManager.kt:3)
	at com.ichi2.anki.CollectionManager$withCol$2.invoke(CollectionManager.kt:1)
	at com.ichi2.anki.CollectionManager$withQueue$2.invokeSuspend(CollectionManager.kt:15)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:12)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:119)
	at kotlinx.coroutines.internal.LimitedDispatcher.run(LimitedDispatcher.kt:13)
	at kotlinx.coroutines.scheduling.TaskImpl.run(Tasks.kt:3)
	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:1)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:15)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:29)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:1)
Caused by: net.ankiweb.rsdroid.BackendException$BackendDbException: DbError { info: "SqlInputError { error: Error { code: Unknown, extended_code: 1 }, msg: \"near \\\"s\\\": syntax error\", sql: \"ATTACH '/storage/emulated/0/Android/data/com.ichi2.anki/cache/export/Life's skill-20230421181612.anki2' AS DST_DB\", offset: 74 }", kind: Other }
	at net.ankiweb.rsdroid.BackendException$BackendDbException$Companion.fromDbError(BackendException.kt:73)
	at net.ankiweb.rsdroid.BackendException$Companion.fromError(BackendException.kt:160)
	at net.ankiweb.rsdroid.BackendKt.unpackResult(Backend.kt:38)
	at net.ankiweb.rsdroid.BackendKt.access$unpackResult(Backend.kt:1)
	at net.ankiweb.rsdroid.Backend$runMethodRaw$1.invoke(Backend.kt:2)
	at net.ankiweb.rsdroid.Backend$runMethodRaw$1.invoke(Backend.kt:1)
	at net.ankiweb.rsdroid.Backend.withBackend(Backend.kt:13)
	at net.ankiweb.rsdroid.Backend.runMethodRaw(Backend.kt:16)
	at anki.backend.GeneratedBackend.runDbCommandProtoRaw(GeneratedBackend.kt:9)
	at anki.backend.GeneratedBackend.runDbCommandProto(GeneratedBackend.kt:26)
	at net.ankiweb.rsdroid.Backend.fullQueryProto(Backend.kt:20)
	at net.ankiweb.rsdroid.database.Session.fullQueryProto(Session.kt:8)
	at net.ankiweb.rsdroid.database.StreamingProtobufSQLiteCursor.<init>(StreamingProtobufSQLiteCursor.kt:21)
	... 20 more
```

* 2.16alpha97
* https://ankidroid.org/acra/app/1/bug/45903/report/031fd479-1af5-4677-8314-67281be1f95c

--- a/AnkiDroid/src/main/java/com/ichi2/anki/export/ActivityExportingDelegate.kt
+++ b/AnkiDroid/src/main/java/com/ichi2/anki/export/ActivityExportingDelegate.kt
@@ -38,6 +38,7 @@ import com.ichi2.anki.dialogs.ExportDialog.ExportDialogListener
 import com.ichi2.anki.dialogs.ExportDialogParams
 import com.ichi2.anki.servicelayer.ScopedStorageService
 import com.ichi2.anki.snackbar.showSnackbar
+import com.ichi2.annotations.NeedsTest
 import com.ichi2.compat.CompatHelper
 import com.ichi2.libanki.AnkiPackageExporter
 import com.ichi2.libanki.Collection
@@ -98,6 +99,7 @@ class ActivityExportingDelegate(private val activity: AnkiActivity, private val 
         }
     }
 
+    @NeedsTest("exporting deck with name containing apostrophe")
     override fun exportColAsApkgOrColpkg(path: String?, includeSched: Boolean, includeMedia: Boolean) {
         val exportPath = getExportFileName(path, "All Decks", includeSched)
 

--- a/AnkiDroid/src/main/java/com/ichi2/libanki/AnkiPackageExporter.kt
+++ b/AnkiDroid/src/main/java/com/ichi2/libanki/AnkiPackageExporter.kt
@@ -143,7 +143,7 @@ open class AnkiExporter(col: Collection, did: DeckId?, val includeSched: Boolean
         // flexible
         dst.close()
         Timber.d("Attach DB")
-        col.db.database.execSQL("ATTACH '$path' AS DST_DB")
+        col.db.database.execSQL("ATTACH ? AS DST_DB", arrayOf(path))
         // copy cards, noting used nids (as unique set)
         Timber.d("Copy cards")
         col.db.database

